### ISSUE TITLE
Synchronization from Vimeo fails

### ISSUE THREAD

**Issue by fflorent** [Chocobozzz/PeerTube#6492]
### Describe the current behavior

Synchronization of videos from vimeo fails with this error:

```
WARNING: [vimeo] The extractor is attempting impersonation, but no impersonate target is available; if you encounter errors, then see  https://github.com/yt-dlp/yt-dlp#impersonation  for information on installing the required dependencies

ERROR: [vimeo] 904507246: This request has been blocked due to its TLS fingerprint. Install a required impersonation dependency if possible, or else if you are okay with compromising your security/cookies, try replacing "https:" with "http:" in the input URL. If you are using a data center IP or VPN/proxy, your IP may be blocked.
```

To solve this problem, impersonation is suggested:
https://github.com/yt-dlp/yt-dlp#impersonation

Binaries for windows, mac and linux seem to already contain the required library. What do you think

### Steps to reproduce

1.
2.
3.

### Additional information

* PeerTube instance:
  * URL: https://skeptikon.fr
  * Version: 6.1.0
  * NodeJS version: v20.15.1
  * Ffmpeg version: 4.3.7-0+deb11u1

* Browser name, version and platforms on which you could reproduce the bug:
* Link to browser console log if relevant:
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):

**Comment by Chocobozzz** [comment]
`yt-dlp_linux` seems to include this library but not `yt-dlp` (https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#recommended)

Can you try to install this library globally on your system to see if it fixes the issue?

**Comment by fflorent** [comment]
I already tested this:
- manually ran `yt-dlp` on a vimeo video, it failed;
- manually ran `yt-dlp_linux`, it suceeded in downloading the same video;

I did not install globally though. You would like to see whether replacing the binary would fix the issue in PeerTube? I could not test that because the wrapper calls the python binary before calling yt-dlp

**Comment by Chocobozzz** [comment]
I meant installing `curl_cffi` globally :) If it doesn't work I'll have to add youtube-dl binaries support to peertube

**Comment by fflorent** [comment]
TL;DR: That seems to work!

I did:
```
pip install "yt-dlp[default,curl-cffi]"
```

And then symlinked the yt-dlp python executable to the path expected by PeerTube.

I used to have these logs:
```
août 11 16:12:57 skeptikon.fr peertube[1528]: [skeptikon.fr:443] 2024-08-11 16:12:57.455 error: Cannot build import for https://vimeo.com/544902990 in channel Horizons {
août 11 16:12:57 skeptikon.fr peertube[1528]:   "err": {
août 11 16:12:57 skeptikon.fr peertube[1528]:     "stack": "ExecaError: Command failed with exit code 1: /usr/bin/python3 /home/yunohost.app/peertube/storage/bin/yt-dlp --dump-json -f 'bestvideo[vcodec!*=av01][vcodec!*=vp9.2]+bestaudio/best[vcodec!*=av01][vcodec!*=vp9.2]/bestvideo[ext=mp4]+bestaudio[ext=m4a]/best' 'https://vimeo.com/544902990'\n\nWARNING: [vimeo] The extractor is attempting impersonation, but no impersonate target is available; if you encounter errors, then see  https://github.com/yt-dlp/yt-dlp#impersonation  for information on installing the required dependencies\nERROR: [vimeo] 544902990: This request has been blocked due to its TLS fingerprint. Install a required impersonation dependency if possible, or else if you are okay with compromising your security/cookies, try replacing \"https:\" with \"http:\" in the input URL. If you are using a data center IP or VPN/proxy, your IP may be blocked.\n    at getFinalError (file:///var/www/peertube/node_modules/execa/lib/return/final-error.js:6:9)\n    at makeError (file:///var/www/peertube/node_modules/execa/lib/return/result.js:108:16)\n    at getAsyncResult (file:///var/www/peertube/node_modules/execa/lib/methods/main-async.js:167:4)\n    at handlePromise (file:///var/www/peertube/node_modules/execa/lib/methods/main-async.js:150:17)\n    at async YoutubeDLCLI.run (file:///var/www/peertube/dist/core/helpers/youtube-dl/youtube-dl-cli.js:141:24)\n    at async YoutubeDLCLI.getInfo (file:///var/www/peertube/dist/core/helpers/youtube-dl/youtube-dl-cli.js:89:22)\n    at async YoutubeDLWrapper.getInfoForDownload (file:///var/www/peertube/dist/core/helpers/youtube-dl/youtube-dl-wrapper.js:23:22)\n    at async skipImport (file:///var/www/peertube/dist/core/lib/sync-channel.js:72:27)\n    at async synchronizeChannel (file:///var/www/peertube/dist/core/lib/sync-channel.js:34:21)\n    at async VideoChannelSyncLatestScheduler.internalExecute (file:///var/www/peertube/dist/core/lib/schedulers/video-channel-sync-latest-scheduler.js:24:13)\n    at async VideoChannelSyncLatestScheduler.execute (file:///var/www/peertube/dist/core/lib/schedulers/abstract-scheduler.js:19:13)",
août 11 16:12:57 skeptikon.fr peertube[1528]:     "message": "Command failed with exit code 1: /usr/bin/python3 /home/yunohost.app/peertube/storage/bin/yt-dlp --dump-json -f 'bestvideo[vcodec!*=av01][vcodec!*=vp9.2]+bestaudio/best[vcodec!*=av01][vcodec!*=vp9.2]/bestvideo[ext=mp4]+bestaudio[ext=m4a]/best' 'https://vimeo.com/544902990'\n\nWARNING: [vimeo] The extractor is attempting impersonation, but no impersonate target is available; if you encounter errors, then see  https://github.com/yt-dlp/yt-dlp#impersonation  for information on installing the required dependencies\nERROR: [vimeo] 544902990: This request has been blocked due to its TLS fingerprint. Install a required impersonation dependency if possible, or else if you are okay with compromising your security/cookies, try replacing \"https:\" with \"http:\" in the input URL. If you are using a data center IP or VPN/proxy, your IP may be blocked.",
août 11 16:12:57 skeptikon.fr peertube[1528]:     "shortMessage": "Command failed with exit code 1: /usr/bin/python3 /home/yunohost.app/peertube/storage/bin/yt-dlp --dump-json -f 'bestvideo[vcodec!*=av01][vcodec!*=vp9.2]+bestaudio/best[vcodec!*=av01][vcodec!*=vp9.2]/bestvideo[ext=mp4]+bestaudio[ext=m4a]/best' 'https://vimeo.com/544902990'",
août 11 16:12:57 skeptikon.fr peertube[1528]:     "command": "/usr/bin/python3 /home/yunohost.app/peertube/storage/bin/yt-dlp --dump-json -f bestvideo[vcodec!*=av01][vcodec!*=vp9.2]+bestaudio/best[vcodec!*=av01][vcodec!*=vp9.2]/bestvideo[ext=mp4]+bestaudio[ext=m4a]/best https://vimeo.com/544902990",
```

I now have these logs, which indicated it seems to work, though no new video is downloaded as none has been published since on Vimeo:

```
août 11 18:09:23 skeptikon.fr peertube[1528]: [skeptikon.fr:443] 2024-08-11 18:09:23.155 info: Fetched 10 candidate URLs for sync channel horizons. {
août 11 18:09:23 skeptikon.fr peertube[1528]:   "targetUrls": [
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/544902990",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/544903766",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/548296687",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/570826043",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/647735985",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/809019651",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/890615737",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/904507246",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/918828477",
août 11 18:09:23 skeptikon.fr peertube[1528]:     "https://vimeo.com/955947025"
août 11 18:09:23 skeptikon.fr peertube[1528]:   ]
août 11 18:09:23 skeptikon.fr peertube[1528]: }
```

**Comment by Chocobozzz** [comment]
Thanks! With https://github.com/Chocobozzz/PeerTube/commit/38dbafcf63cf780f50e8c1b073839dce2a7becf7 we'll be able to specify a yt-dlp executable in the config in the next PeerTube version (with `python_path: null`)
